server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -2011,17 +2011,27 @@ public static String createEmptyTranslog(
         final long primaryTerm,
         @Nullable final String translogUUID,
         @Nullable final ChannelFactory factory
+    ) throws IOException {
+        return createEmptyTranslog(location, shardId, initialGlobalCheckpoint, primaryTerm, translogUUID, factory, 1);
+    }
+
+    public static String createEmptyTranslog(
+        final Path location,
+        final ShardId shardId,
+        final long initialGlobalCheckpoint,
+        final long primaryTerm,
+        @Nullable final String translogUUID,
+        @Nullable final ChannelFactory factory,
+        final long generation
```

Comment:
taking generation while creating empty translog doesn't look right. Can we start with generation of 1 and use `translogUUID` from existing checkpoint .

Reply:
> use translogUUID from existing checkpoint .

We are using `translogUUID` from existing checkpoint

> Can we start with generation of 1

Using generation 1 each time may break things. For example, metadata for remote segment store and remote translog uses `generation` in its name which is used in sorting the metadata files and fetching the latest file. Also, as I understand, generation is supposed to be strictly increasing and not sure if there are any other flows in OpenSearch that depends on generation to be always increasing.